transparent/rvsa.go

```diff
@@ -0,0 +1,207 @@
+/* Copyright 2019 Freerware
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package transparent
+
+import (
+	"net/http"
+
+	"github.com/freerware/negotiator/internal"
+	"github.com/freerware/negotiator/internal/header"
+	"github.com/freerware/negotiator/representation"
+)
+
+// rvsa1 represents the Remote Variant Selection Algorithm 1.0 as
+// defined in RFC2296. This algorithm is leveraged in remote variant
+// selection within transparent content negotiation.
+type rvsa1 struct{}
+
+// RVSA1 provides the Remote Variant Selection Algorithm 1.0 as
+// defined in RFC2296.
+func RVSA1() representation.Chooser {
+	return rvsa1{}
+}
+
+// Choose determines the 'best' representation from the provided set.
+func (c rvsa1) Choose(
+	r *http.Request, reps ...representation.Representation) (representation.Representation, error) {
+
+	var a header.Accept
+	//var ae header.AcceptEncoding
+	var al header.AcceptLanguage
+	var ac header.AcceptCharset
+	var af header.AcceptFeatures
+	var err error
+
+	accept, hasAccept := r.Header["Accept"]
+	if a, err = header.NewAccept(accept); err != nil {
+		return nil, err
+	}
+
+	//acceptEncoding, hasAcceptEncoding := r.Header["Accept-Encoding"]
+	//if ae, err = header.NewAcceptEncoding(acceptEncoding); err != nil {
+	//	return nil, err
+	//}
+
+	acceptLanguage, hasAcceptLanguage := r.Header["Accept-Language"]
+	if al, err = header.NewAcceptLanguage(acceptLanguage); err != nil {
+		return nil, err
+	}
+
+	acceptCharset, hasAcceptCharset := r.Header["Accept-Charset"]
+	if ac, err = header.NewAcceptCharset(acceptCharset); err != nil {
+		return nil, err
+	}
+
+	// add isEmpty() to each header implementation?
```

Comment:
adding `.IsEmpty()` for each header type would eliminate the need to have these boolean variables. however, their constructor functions return defaults.

Reply:
done!